### ISSUE TITLE
please add the ability to choose which apps can show notifications

### ISSUE THREAD

**Issue by coffeeking** [mate-desktop/mate-notification-daemon#99]
This is another feature request. It would be neat if mate could turn application's notifications on and off, similar to gnome shell, but much more accessibly than gnome shell. I'd be willing to help implement this once my grasp of c is better.

**Comment by monsta** [comment]
Interesting, I didn't know gnome shell has such a feature. So it's like a blacklist to inhibit notifications from selected apps?

**Comment by coffeeking** [comment]
hi
Yup. It's a dialog with a list of apps that show notifications (I'm not
sure how this is generated) along with options to manipulate which ones
show up on the lock screen. An ability to do this in a display manager
agnostic way would be nice, but I don't underlying standards enough to
know whether this is easily doable. I suspect not or you guys would have
done it already.

monsta wrote:

**Comment by monsta** [comment]
Yes, currently I have no idea how to implement this, but we might try it someday :smile:

**Comment by aoreucro** [comment]
I'm not specifically interested in this feature, but I may be able to provide some useful information ...

Scroll down to "org.freedesktop.Notifications.Notify":
https://developer.gnome.org/notification-spec/#commands

The first parameter is the app_name.

In dbus-monitor, this looks like this:
```
method call time=1482081219.099802 sender=:1.236199 -> destination=org.freedesktop.Notifications serial=18 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=Notify
   string "FileZilla"
   uint32 0
   string "filezilla"
   string "Transfers finished"
   string "All files have been successfully transferred"
   array [
   ]
   array [
      dict entry(
         string "category"
         variant             string "transfer.complete"
      )
   ]
   int32 -1
```

**Comment by coffeeking** [comment]
hi

This would be a great feature to have. Gnome has this feature, although 
not all apps support it, but for those that do it's really handy. 
Windows 10 also has this feature and it works rather well. If this 
feature does get implemented, I'd suggested a simple list view with 
check boxes, or maybe a list view with the names of the apps, and if the 
enter key is pressed a little dialog box opens up, showing a short 
description, and then an on or off button. Gnome's implementation has 
some accessibility issues, windows 10's does not. I'm not sure how 
helpful i can be, but this would be a great feature for mate to have.

**Comment by dmeziere** [comment]
If i understand well, it could stop the "weak battery" notification from popping every 5 min ? My battery is dead for years, now. It would be so nice to stop that popup !
